### DISCUSSION TITLE
how to Compute the eigenvalues and eigenvectors of ndarray/hidden layer?

### DISCUSSION THREAD

**Discussion by weihua04** [apache/incubator-mxnet#13944] [Q&A]
i want to compute the eigenvalues and eigenvectors of a hidden layer using mxnet, although i just know numpy can do this, i'm not sure mxnet have any methods to compute the eigenvalues and eigenvectors of a hidden layer,

thank u very much

**Comment by piyushghai** [comment]
@weihua04 Thanks for posting this. This seems like a question to me and this can get more attention on the MXNet discussion forum (https://discuss.mxnet.io), where it will get a wider audience and allow others to learn as well.
Thanks!

@mxnet-label-bot Add[Question]

**Comment by ChaiBapchya** [comment]
@weihua04 
Our discussion forum is a good place for these questions. But regardless, here's the answer
On our documentation - https://mxnet.incubator.apache.org/api/python/ndarray/linalg.html
`mxnet.ndarray.linalg.syevd` seems to be the function for your usecase.

Documentation on the website also gives the following 2 examples
```
// Single symmetric eigendecomposition
A = [[1., 2.], [2., 4.]]
U, L = syevd(A)
U = [[0.89442719, -0.4472136],
     [0.4472136, 0.89442719]]
L = [0., 5.]

// Batch symmetric eigendecomposition
A = [[[1., 2.], [2., 4.]],
     [[1., 2.], [2., 5.]]]
U, L = syevd(A)
U = [[[0.89442719, -0.4472136],
      [0.4472136, 0.89442719]],
     [[0.92387953, -0.38268343],
      [0.38268343, 0.92387953]]]
L = [[0., 5.],
     [0.17157288, 5.82842712]]
```
where
`U` is orthonormal matrix of eigenvectors,  `L` is vector of eigenvalues
Hope this helps.

**Comment by weihua04** [comment]
@ChaiBapchya ,thank u very much for reply. I'll post question  on the MXNet discussion forum later.
      i have seen mxnet.ndarray.linalg.syevd ,but it only support the symmetric matrix, if the matrix is asymmetric how to solve?
      i resolved this problem by using ".asnumpy()", and then using " .eig(T)" support by numpy ,but i found it run really slow.
     do u have any good idea to solve it?
